fluxette/js/flux/stores.js

```diff
@@ -0,0 +1,23 @@
+import { Store } from 'fluxette';
+import { CART, API } from './types';
+import { splicefilter } from '../util';
+
+let products = Store([], {
+    [API.PRODUCTS.DONE]: (products, action) => action.products,
+    [CART.ADD]: (products, action) => {
+        let { id } = action.product;
+        let [left, product, right] = splicefilter(products, p => p.id === id);
+        return [...left, { ...product, inventory: product.inventory - 1 }, ...right];
```

Comment:
@edge after the bought of all items there should be `inventory: 0` in console product but now it stays the same - 

![](https://dl.dropboxusercontent.com/u/100463011/%D0%A1%D0%BD%D0%B8%D0%BC%D0%BE%D0%BA%20%D1%8D%D0%BA%D1%80%D0%B0%D0%BD%D0%B0%202015-07-23%20%D0%B2%2010.50.19.png)